src/varname.jl

```diff
@@ -671,30 +670,14 @@ function _parse_obj_optic(ex)
     obj, optic
 end
 
-# Accessors doesn't have the same support for interpolation, so copy and modify Setfield's parsing functions
-is_interpolation(x) = x isa Expr && x.head == :$
-
-function _parse_obj_optics_composite(lensexprs::Vector)
-    if isempty(lensexprs)
-        return esc(:_), ()
-    else
-        obj, outermostlens = _parse_obj_optics(lensexprs[1])
-        innerlenses = map(lensexprs[2:end]) do innerex
-            o, lens = _parse_obj_optics(innerex)
-            @assert o == esc(:_)
-            lens
-        end
-        return obj, (outermostlens, innerlenses...)
-    end
-end
-
+# Accessors doesn't have the same support for interpolation
+# so this function is copied and altered from `Setfield._parse_obj_lens`
 function _parse_obj_optics(ex)
-    if @capture(ex, ∘(opticsexprs__))
-        return _parse_obj_optics_composite(opticsexprs)
-    elseif is_interpolation(ex)
+    if ex isa Expr && ex.head == :$
         @assert length(ex.args) == 1
         return esc(:_), (esc(ex.args[1]),)
-    elseif @capture(ex, front_[indices__])
+    elseif Meta.isexpr(ex, :ref)
```

Comment:
The next line will fail if there is not at least one argument:
```suggestion
    elseif Meta.isexpr(ex, :ref) && !isempty(ex.args)
```